openpathsampling/experimental/simstore/storage.py

```diff
@@ -587,20 +585,33 @@ def __iter__(self):
 
     def __getitem__(self, item):
         len_self = len(self)
-        if not (-len_self <= item < len_self):
-            raise IndexError("table index out of range")
-
-        if item < 0:
-            item += len(self)
+        if item is slice:
+            unpack_return = False
+            items = (i for i in range(*item.indices(len_self)))
+        elif type(item) is int:
+            unpack_return = True
+            if not (-len_self <= item < len_self):
+                raise IndexError("table index out of range")
+
+            if item < 0:
+                item += len(self)
+            items = [item]
+        else:
+            raise TypeError("Only acces via slice or int allowed, "
+                            f"got type {type(item)}.")
 
-        row = self.storage.backend.table_get_item(self.table, item)
-        return self.storage.load([row.uuid])[0]
+        rows = [self.storage.backend.table_get_item(self.table, item).uuid
+                for item in items]
+        if unpack_return:
+            return self.storage.load(rows)[0]
+        else:
+            return self.storage.load(rows)
 
     def __len__(self):
         return self.storage.backend.table_len(self.table)
 
     def cache_all(self):
-        old_blocksize = self.iter_block_size
+        # old_blocksize = self.iter_block_size
```

Comment:
@dwhswenson this variable was unused, but I didn't know if you had any future plans with this. If not, this line can be removed

Reply:
It looks to me like I forgot to reset the block size. See suggestion for this method (separate comment in review because of GitHub internals). I don't think this has any significant effect in practice, but it does make sense for `cache_all` to *temporarily* change the block size for iteration into "load everything at once." As the code stands, it permanently makes that change.

I don't think it matters in practice, because once everything is cached, the value of `iter_block_size` shouldn't need to be used again. But the cleaner way to structure the code is to not have non-obvious side effects. Thanks for catching this!